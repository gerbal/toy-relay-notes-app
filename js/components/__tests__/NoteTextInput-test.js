import React from 'react';
import NoteTextInput from '../NoteTextInput';

import { mount } from 'enzyme';
import { StyleSheetTestUtils } from 'aphrodite';

// Todo: Put these methods in testSetup
beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection();
});
afterEach(() => {
  return new Promise(resolve => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
    return process.nextTick(resolve);
  });
});

// The only component not dependant on Relay.
it('renders correctly', async () => {
  const onSave = jest.fn();
  const wrapper = mount(<NoteTextInput onSave={onSave} />);

  expect(wrapper).toMatchSnapshot();
});

it('calls onSave when receiving a Enter keypress', async () => {
  const onSave = jest.fn();
  const wrapper = mount(<NoteTextInput onSave={onSave} />);

  wrapper.find('input').simulate('change', { target: { value: 'text' } });

  expect(wrapper).toMatchSnapshot();
  wrapper.find('input').simulate('keyDown', { keyCode: 13 });
  expect(onSave).toHaveBeenCalled();
  expect(wrapper).toMatchSnapshot();
});

it('calls onSave onBlur', async () => {
  const onSave = jest.fn();
  const wrapper = mount(<NoteTextInput onSave={onSave} commitOnBlur />);

  wrapper.find('input').simulate('change', { target: { value: 'blur' } });

  expect(wrapper).toMatchSnapshot();
  wrapper.find('input').simulate('blur');
  expect(onSave).toHaveBeenCalled();
  expect(wrapper).toMatchSnapshot();
});
