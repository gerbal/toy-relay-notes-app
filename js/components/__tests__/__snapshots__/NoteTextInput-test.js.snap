// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls onSave onBlur 1`] = `
<NoteTextInput
  commitOnBlur={true}
  onSave={[Function]}
>
  <div
    className="chatInput_1cxtnt9"
  >
    <input
      className="textInput_14xfth8"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      value="blur"
    />
  </div>
</NoteTextInput>
`;

exports[`calls onSave onBlur 2`] = `
<NoteTextInput
  commitOnBlur={true}
  onSave={[Function]}
>
  <div
    className="chatInput_1cxtnt9"
  >
    <input
      className="textInput_14xfth8"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      value=""
    />
  </div>
</NoteTextInput>
`;

exports[`calls onSave when receiving a Enter keypress 1`] = `
<NoteTextInput
  commitOnBlur={false}
  onSave={[Function]}
>
  <div
    className="chatInput_1cxtnt9"
  >
    <input
      className="textInput_14xfth8"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      value="text"
    />
  </div>
</NoteTextInput>
`;

exports[`calls onSave when receiving a Enter keypress 2`] = `
<NoteTextInput
  commitOnBlur={false}
  onSave={[Function]}
>
  <div
    className="chatInput_1cxtnt9"
  >
    <input
      className="textInput_14xfth8"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      value=""
    />
  </div>
</NoteTextInput>
`;

exports[`renders correctly 1`] = `
<NoteTextInput
  commitOnBlur={false}
  onSave={[Function]}
>
  <div
    className="chatInput_1cxtnt9"
  >
    <input
      className="textInput_14xfth8"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      value=""
    />
  </div>
</NoteTextInput>
`;
